### PR TITLE
USB Device Mode Driver Support for RX65N

### PR DESCRIPTION
**Pull request by anjana-tel**
## Summary
Added Support for USB Device Mode driver on RX65N

## Impact
Porting the RX65N Device Mode driver

### CODE REVIEW

File: arch/renesas/src/rx65n/Kconfig [hunk 1]
```diff
@@ -610,6 +610,21 @@ config RX65N_RIIC2_RCV_IN_BYTE_UNITS
...
+config RX65N_USBDEV 
+	bool "USB0 Device"
+	default n
+	depends on USBDEV
+
+if RX65N_USBDEV
+
+config RX65N_TEST_INTEP
+    bool "Testing RX65N Interrupt endpoint"
+	default n
```
Comment (acassis) [inline arch/renesas/src/rx65n/Kconfig:623]: bool and default n aren't aligned

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 2]
```diff
@@ -594,6 +594,855 @@
...
+#define USB_WRITING                     (0x0002u)   /* Write continues */
+#define USB0_CFIFO8                     (USB0.CFIFO.BYTE.L)
+#define USB0_D0FIFO8                    (USB0.D0FIFO.BYTE.L)
+#define USB0_D1FIFO8                    (USB0.D1FIFO.BYTE.L)
+#define USB0_CFIFO16                    (USB0.CFIFO.WORD)
+#define USB0_D0FIFO16                   (USB0.D0FIFO.WORD)
+#define USB0_D1FIFO16                   (USB0.D1FIFO.WORD)
+#define USB_WRITEEND                    (0x0000u)     
+#define USB_CTRL_END                    (0u)
+#define     USB_BREQUEST                (0xFF00u)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:1029]: this define is unaligned

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 3]
```diff
@@ -594,6 +594,855 @@
...
 
+/* USB Related definitions */
+
+#define RX65N_NUSBHOST   1
+
+/* USB Registers */
+
+/* USB Peripheral base address */
+
+#define RX65N_MSTPCRB_START_STOP_USB    (1<<19)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:605]: All these "X<<Y" should be replaced with "X << Y"

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 4]
```diff
@@ -594,6 +594,855 @@
...
+/* Different USB registers with corresponding offset */
+
+/* USB System Configuration register and its bit fields */
+
+#define RX65N_USB_SYSCFG                ((volatile short *) (RX65N_USB_BASE + 0x0000UL))
+#define RX65N_USB_SYSCFG_SCKE           (1U<<10)
+#define RX65N_USB_SYSCFG_DCFM           (1U<<6)
+#define RX65N_USB_SYSCFG_DRPD           (1U<<5)
+#define RX65N_USB_SYSCFG_DPRPU          (1U<<4)
+#define RX65N_USB_SYSCFG_USBE           (1U<<0)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:617]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 5]
```diff
@@ -594,6 +594,855 @@
...
+
+#define RX65N_USB_SYSSTS0               ((volatile short *) (RX65N_USB_BASE + 0x0004UL))
+#define USB_FS_JSTS                     (0x0001u)   /* Full-Speed J State */
+#define USB_LNST                        (0x0003u)   /* b1-0: D+, D- line status */
+#define RX65N_USB_SYSSTS0_LNST          (3)
+#define RX65N_USB_SYSSTS0_IDMON         (1U<<2)
+#define RX65N_USB_SYSSTS0_SOFEA         (1U<<5)
+#define RX65N_USB_SYSSTS0_HTACT         (1U<<6)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:625]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 6]
```diff
@@ -594,6 +594,855 @@
...
+
+#define RX65N_USB_DVSTCTR0                      ((volatile short *) (RX65N_USB_BASE + 0x0008UL))
+#define RX65N_USB_DVSTCTR0_HNPBTOA              (1U<<11)
+#define RX65N_USB_DVSTCTR0_EXICEN               (1U<<10)
+#define RX65N_USB_DVSTCTR0_VBUSEN               (1U<<9)
+#define RX65N_USB_DVSTCTR0_WKUP                 (1U<<8)
+#define RX65N_USB_DVSTCTR0_RWUPE                (1U<<7)
+#define RX65N_USB_DVSTCTR0_USBRST               (1U<<6)
+#define RX65N_USB_DVSTCTR0_RESUME               (1U<<5)
+#define RX65N_USB_DVSTCTR0_UACT                 (1U<<4)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:688]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 7]
```diff
@@ -594,6 +594,855 @@
...
+#define RX65N_USB_D0FIFO            ((volatile short *) (RX65N_USB_BASE + 0x0018UL))
+
+/*      USB D1FIFO Port Register and its bit fields */
+
+#define RX65N_USB_D1FIFO            ((volatile short *) (RX65N_USB_BASE + 0x001cUL))
+
+/*      USB CFIFO Port Select Register and its bit fields */
+
+#define RX65N_USB_CFIFOSEL              ((volatile short *) (RX65N_USB_BASE + 0x0020UL))
+#define RX65N_USB_CFIFOSEL_RCNT         (1U<<15)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:710]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 8]
```diff
@@ -594,6 +594,855 @@
...
+#define USB_RCNT                        (RX65N_USB_CFIFOSEL_RCNT)
+#define RX65N_USB_CFIFOSEL_REW          (1U<<14)
+#define RX65N_USB_CFIFOSEL_MBW_8        (0U<<10)
+#define RX65N_USB_CFIFOSEL_MBW_16       (1U<<10)
+#define RX65N_USB_CFIFOSEL_BIGEND       (1U<<8)
+#define RX65N_USB_CFIFOSEL_ISEL         (1U<<5)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:716]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 9]
```diff
@@ -594,6 +594,855 @@
...
+#define RX65N_USB_CFIFOSEL_CURPIPE_MASK (0xfU)
+#define USB_CURPIPE                     (RX65N_USB_CFIFOSEL_CURPIPE_MASK)
+
+/*      USB CFIFO Port Control Register */
+
+#define RX65N_USB_CFIFOCTR              ((volatile short *) (RX65N_USB_BASE + 0x0022UL))
+
+/* Common bit field values for CFIFOCTR, D0FIFOCTR and D1FIFOCTR registers */
+
+#define RX65N_USB_FIFOCTR_BVAL          (1U<<15)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:727]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 10]
```diff
@@ -594,6 +594,855 @@
...
+#define USB_BVAL                        (RX65N_USB_FIFOCTR_BVAL)
+#define RX65N_USB_FIFOCTR_BCLR          (1U<<14)
+#define RX65N_USB_FIFOCTR_FRDY          (1U<<13)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:730]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 11]
```diff
@@ -594,6 +594,855 @@
...
+#define RX65N_USB_FIFOCTR_DTLN          (0xfff)
+
+/*      USB D0FIFO and D1FIFO port select and control registers */
+
+#define RX65N_USB_D0FIFOSEL             ((volatile short *) (RX65N_USB_BASE + 0x0028UL))
+#define RX65N_USB_D0FIFOSEL_MBW_16      (1U<<10)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:736]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 12]
```diff
@@ -594,6 +594,855 @@
...
+#define RX65N_USB_D0FIFOCTR             ((volatile short *) (RX65N_USB_BASE + 0x002aUL))
+#define RX65N_USB_D1FIFOSEL             ((volatile short *) (RX65N_USB_BASE + 0x002cUL))
+#define RX65N_USB_D1FIFOSEL_MBW_16      (1U<<10)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:739]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 13]
```diff
@@ -594,6 +594,855 @@
...
+#define USB_TRAN_CONT                       (0x00u)
+#define USB_TRAN_END                        (0x80u)
+
+/* USB common bit fields for D0 and D1 FIFO select register */
+
+#define RX65N_USB_DFIFOSEL_RCNT         (1U<<15)
+#define RX65N_USB_DFIFOSEL_REW          (1U<<14)
+#define RX65N_USB_DFIFOSEL_DCLRM        (1U<<13)
+#define RX65N_USB_DFIFOSEL_DREQE        (1U<<12)
+#define RX65N_USB_DFIFOSEL_MBW_8        (0U<<10)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:800]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 14]
```diff
@@ -594,6 +594,855 @@
...
+#define USB_MBW_8                       (RX65N_USB_DFIFOSEL_MBW_8)
+#define RX65N_USB_DFIFOSEL_MBW_16       (1U<<10)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:802]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 15]
```diff
@@ -594,6 +594,855 @@
...
+#define USB_MBW_16                      (RX65N_USB_DFIFOSEL_MBW_16)
+#define USB0_CFIFO_MBW                  (USB_MBW_16)
+#define USB0_D0FIFO_MBW                 (USB_MBW_16)
+#define USB0_D1FIFO_MBW                 (USB_MBW_16)
+#define RX65N_USB_DFIFOSEL_BIGEND       (1U<<8)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:807]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 16]
```diff
@@ -594,6 +594,855 @@
...
+#define RX65N_USB_INTENB0_BEMPE         (1U<<10)
+#define RX65N_USB_INTENB0_BRDYE         (1U<<8)
+#define RX65N_USB_INTENB0_VBSE          (1U<<15)
+#define RX65N_USB_INTENB0_RSME          (1U<<14)
+#define RX65N_USB_INTENB0_SOFE          (1U<<13)
+#define RX65N_USB_INTENB0_DVSE          (1U<<12)
+#define RX65N_USB_INTENB0_CTRE          (1U<<11)
+#define RX65N_USB_INTENB0_BEMPE         (1U<<10)
+#define RX65N_USB_INTENB0_NRDYE         (1U<<9)
+#define RX65N_USB_INTENB0_BRDYE         (1U<<8)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:822]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 17]
```diff
@@ -594,6 +594,855 @@
...
+
+#define RX65N_USB_INTENB1               ((volatile short *) (RX65N_USB_BASE + 0x0032UL))
+#define RX65N_USB_INTENB1_OVRCRE        (1U<<15)
+#define RX65N_USB_INTENB1_BCHGE         (1U<<14)
+#define RX65N_USB_INTENB1_DTCHE         (1U<<12)
+#define RX65N_USB_INTENB1_ATTCHE        (1U<<11)
+
+#define RX65N_USB_INTENB1_EOFERRE       (1U<<6)
+#define RX65N_USB_INTENB1_SIGNE         (1U<<5)
+#define RX65N_USB_INTENB1_SACKE         (1U<<4)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:834]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 18]
```diff
@@ -594,6 +594,855 @@
...
+
+#define RX65N_USB_SOFCFG_EDGESTS        (1U<<4)
+
+/*      USB Interrupt Status Register 0 and its bit fields      */
+
+#define RX65N_USB_INTSTS0               ((volatile short *) (RX65N_USB_BASE + 0x0040UL))
+#define RX65N_USB_INTSTS0_VBINT         (1U<<15)
+#define RX65N_USB_INTSTS0_RESM          (1U<<14)
+#define RX65N_USB_INTSTS0_SOFR          (1U<<13)
+#define RX65N_USB_INTSTS0_DVST          (1U<<12)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:963]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 19]
```diff
@@ -594,6 +594,855 @@
...
+#define USB_DVSQ                        (0x0070u)   /* b6-4: Device state */
+#define RX65N_USB_INTSTS0_CTRT          (1U<<11)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:965]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 20]
```diff
@@ -594,6 +594,855 @@
...
+#define USB_DS_DFLT                     (0x0010u)   /* Default */
+#define USB_DS_POWR                     (0x0000u)   /* Powered */
+
+#define RX65N_USB_INTSTS0_CTRT          (1U<<11)
+#define RX65N_USB_INTSTS0_BEMP          (1U<<10)
+#define RX65N_USB_INTSTS0_NRDY          (1U<<9)
+#define RX65N_USB_INTSTS0_BRDY          (1U<<8)
+#define RX65N_USB_INTSTS0_VBSTS         (1U<<7)
+#define RX65N_USB_INTSTS0_VALID         (1U<<3)
+#define RX65N_USB_INTSTS0_DVSQ_MASK     (7<<4)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:990]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 21]
```diff
@@ -594,6 +594,855 @@
...
+/*      USB Interrupt Status Register 0 and its bit fields      */
+
+#define RX65N_USB_INTSTS1               ((volatile short *) (RX65N_USB_BASE + 0x0042UL))
+#define RX65N_USB_INTSTS1_OVRCRE        (1U<<15)
+#define RX65N_USB_INTSTS1_BCHG          (1U<<14)
+#define RX65N_USB_INTSTS1_DTCH          (1U<<12)
+#define RX65N_USB_INTSTS1_ATTCH         (1U<<11)
+#define RX65N_USB_INTSTS1_EOFERR        (1U<<6)
+#define RX65N_USB_INTSTS1_SIGN          (1U<<5)
+#define RX65N_USB_INTSTS1_SACK          (1U<<4)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:1072]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 22]
```diff
@@ -594,6 +594,855 @@
...
+#define RX65N_USB_INTSTS1_ALL_CLEAR     (0U)
+
+/*      USB DCP Control Register and its bit fields     */
+
+#define RX65N_USB_DCPCTR                ((volatile short *) (RX65N_USB_BASE + 0x0060UL))
+#define RX65N_USB_DCPCTR_BSTS           (1U<<15)
+#define RX65N_USB_DCPCTR_SUREQ          (1U<<14)
+#define RX65N_USB_DCPCTR_SUREQCLR       (1U<<11)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:1080]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 23]
```diff
@@ -594,6 +594,855 @@
...
+#define USB_SUREQCLR                    (RX65N_USB_DCPCTR_SUREQCLR)
+#define RX65N_USB_DCPCTR_SQCLR          (1U<<8)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:1082]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 24]
```diff
@@ -594,6 +594,855 @@
...
+#define USB_SQCLR                       (RX65N_USB_DCPCTR_SQCLR)
+#define RX65N_USB_DCPCTR_SQSET          (1U<<7)
+#define RX65N_USB_DCPCTR_SQMON          (1U<<6)
+#define RX65N_USB_DCPCTR_PBUSY          (1U<<5)
+#define RX65N_USB_DCPCTR_CCPL           (1U<<2)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:1087]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 25]
```diff
@@ -594,6 +594,855 @@
...
+/* USB Pipe 1 to 9 control register bit fields */
+
+#define RX65N_USB_PIPECTR_BSTS          (1U<<15)
+#define RX65N_USB_PIPECTR_INBUFM        (1U<<14)
+#define RX65N_USB_PIPECTR_ATREPM        (1U<<10)
+#define RX65N_USB_PIPECTR_ACLRM         (1U<<9)
+#define RX65N_USB_PIPECTR_SQCLR         (1U<<8)
+#define RX65N_USB_PIPECTR_SQSET         (1U<<7)
+#define RX65N_USB_PIPECTR_SQMON         (1U<<6)
+#define RX65N_USB_PIPECTR_PBUSY         (1U<<5)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:1115]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 26]
```diff
@@ -594,6 +594,855 @@
...
+#define RX65N_USB_PIPECTR_PID_MASK      (3)
+#define RX65N_USB_PIPECTR_PIDNAK        (0)
+#define RX65N_USB_PIPECTR_PIDBUF        (1)
+#define RX65N_USB_PIPECTR_PIDSTALL      (2)
+#define RX65N_USB_PIPECTR_PIDSTALL2     (3)
+#define RX65N_USB_PIPECTR_DATA1         (1U<<7)
+#define RX65N_USB_PIPECTR_DATA0         (1U<<8)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:1122]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 27]
```diff
@@ -594,6 +594,855 @@
...
+#define RX65N_USB_PIPE3TRN                      ((volatile short *) (RX65N_USB_BASE + 0x009aUL))
+#define RX65N_USB_PIPE4TRE                      ((volatile short *) (RX65N_USB_BASE + 0x009cUL))
+#define RX65N_USB_PIPE4TRN                      ((volatile short *) (RX65N_USB_BASE + 0x009eUL))
+#define RX65N_USB_PIPE5TRE                      ((volatile short *) (RX65N_USB_BASE + 0x00a0UL))
+#define RX65N_USB_PIPE5TRN                      ((volatile short *) (RX65N_USB_BASE + 0x00a2UL))
+
+/* USB PIPE 1 to 5 Transaction Counter Enable register bit fields       */
+
+#define RX65N_USB_PIPETRE_TRENB         (1U<<9)
+#define RX65N_USB_PIPETRE_TRCLR         (1U<<8)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:1142]: Ditto
Please align these columns

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 28]
```diff
@@ -594,6 +594,855 @@
...
+
+/*      USB Device Address 0 to 5 Configuration Register */
+
+#define RX65N_USB_DEVADD0                   ((volatile short *) (RX65N_USB_BASE + 0x00d0UL))
+#define RX65N_USB_DEVADD1                   ((volatile short *) (RX65N_USB_BASE + 0x00d2UL))
+#define RX65N_USB_DEVADD2                   ((volatile short *) (RX65N_USB_BASE + 0x00d4UL))
+#define RX65N_USB_DEVADD3                   ((volatile short *) (RX65N_USB_BASE + 0x00d6UL))
+#define RX65N_USB_DEVADD4                   ((volatile short *) (RX65N_USB_BASE + 0x00d8UL))
+#define RX65N_USB_DEVADD5                   ((volatile short *) (RX65N_USB_BASE + 0x00daUL))
+#define RX65N_USB_DEVSPD                    (3<<6)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:1152]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 29]
```diff
@@ -594,6 +594,855 @@
...
+#define USB_DEVICE_9                        (0x9000u) /* Device address 9 */
+#define USB_DEVICE_A                        (0xa000u) /* Device address A */
+#define USB_NODEVICE                        (0xf000u) /* No device */
+#define USB_DEVADDRBIT                      (12u)
+
+/* Device Address bit fields    */
+
+#define RX65N_USB_DEVADD_SPEED_LOW      (1U<<6)
+#define RX65N_USB_DEVADD_SPEED_FULL     (2U<<6)
+#define RX65N_USB_DEVADD_SPEED_HIGH     (3U<<6)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:1173]: Ditto
Please align these columns

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 30]
```diff
@@ -594,6 +594,855 @@
...
+
+#define RX65N_USB_PHYSLEW                       ((volatile int *) (RX65N_USB_BASE + 0x00f0UL))
+
+/* PHY Cross Point Adjustment, note that Hardware Manual to be
+ * updated(0xe->0x5)
+ */
+#define RX65N_USB_PHYSLEW_SLEW_SLEWR00  (1U<<0)
+#define RX65N_USB_PHYSLEW_SLEW_SLEWR01  (1U<<1)
+#define RX65N_USB_PHYSLEW_SLEW_SLEWF00  (1U<<2)
+#define RX65N_USB_PHYSLEW_SLEW_SLEWF01  (1U<<3)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:1185]: Ditto
Please align these columns

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 31]
```diff
@@ -594,6 +594,855 @@
...
+/*      USB BEMP Interrupt Enable Register      */
+
+#define RX65N_USB_BEMPENB               ((volatile short *) (RX65N_USB_BASE + 0x003aUL))
+
+/*      USB Frame Number Register and its bit fields    */
+
+#define RX65N_USB_FRMNUM                ((volatile short *) (RX65N_USB_BASE + 0x004cUL))
+#define RX65N_USB_FRMNUM_OVRN           (1U<<15)
+#define RX65N_USB_FRMNUM_CRCE           (1U<<14)
+#define RX65N_USB_FRMNUM_FRNM_MASK      (0x7ffU)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:1207]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 32]
```diff
@@ -594,6 +594,855 @@
...
+#define RX65N_USB_PIPECFG                   ((volatile short *) (RX65N_USB_BASE + 0x0068UL))
+#define RX65N_USB_PIPECFG_TYPE_MASK         (0xc000)
+#define RX65N_USB_PIPECFG_TYPE_BIT_USED     (0)
+#define RX65N_USB_PIPECFG_TYPE_BULK         (1U<<14)
+#define RX65N_USB_PIPECFG_TYPE_INTERRUPT    (2U<<14)
+#define RX65N_USB_PIPECFG_TYPE_ISOCHRONOUS  (3U<<14)
+#define RX65N_USB_PIPECFG_BFRE              (1U<<10)
+#define RX65N_USB_PIPECFG_DBLB              (1U<<9)
+#define RX65N_USB_PIPECFG_SHTNAK            (1U<<7)
+#define RX65N_USB_PIPECFG_DIR               (1U<<4)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:1225]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 33]
```diff
@@ -594,6 +594,855 @@
...
+#define RX65N_USB_PIPECFG_EPNUM_MASK        (0xfU)
+
+#define RX65N_USB_PIPEMAXP              ((volatile short *) (RX65N_USB_BASE + 0x006cUL))
+#define RX65N_USB_PIPEMAXP_DEVSELMASK   (0xfU<<12)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:1229]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 34]
```diff
@@ -594,6 +594,855 @@
...
+#define USB_CFG_CNTMDON                 (0x0100u)
+#define USB_CFG_CNTMDOFF                (0x0000u)
+#define USB_CFG_DBLB                    (USB_CFG_DBLBON)
+#define USB_DIR_P_IN                    (0x0010u)   /* PERI IN */
+#define USB_DIR_H_IN                    (0x0000u)
+#define USB_SHTNAKFIELD                 (0x0080u)   /* Transfer end NAK */
+#define USB_DIR_P_OUT                   (0x0000u)   /* PERI OUT */
+#define USB_BRDY                        (0x0100u)   /* b8: Buffer ready interrupt */
+#define BRDYSTS_MASK                    (0x03FFu)   /* BRDYSTS Reserved bit mask */
+#define RX65N_USB_INTSTS0_NRDY          (1U<<9)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:1323]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 35]
```diff
@@ -594,6 +594,855 @@
...
+#define RX65N_PIPENUM_WRITE             (1)
+#define RX65N_USB_MAXP                  (64)
+#define RX65N_USBI0_SOURCE              (0x3eu)
+#define RX65N_USBI0_PRIORITY            (0x0f)
+#define RX65N_PHYSLEW_VALUE             (0x5)
+
+#define RX65N_USB_PFKUSB_ENABLED        (1U<<4)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:1330]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 36]
```diff
@@ -594,6 +594,855 @@
...
+#define RX65N_USB_SYSSTS0_LNST_J_STATE_LS            (2U)
+#define RX65N_USB_PLLSTA_PLLLOCK                     (1U<<0)
+#define RX65N_USB_PHYSET_HSEB                        (1U<<15)
+#define RX65N_USB_PHYSET_REPSTART                    (1U<<11)
+#define RX65N_USB_PHYSET_REPSEL                      (1U<<8)
+#define RX65N_USB_PHYSET_CLKSEL_1                    (1U<<5)
+#define RX65N_USB_PHYSET_CLKSEL_0                    (1U<<4)
+#define RX65N_USB_PHYSET_CDPEN                       (1U<<3)
+#define RX65N_USB_PHYSET_PLLRESET                    (1U<<1)
+#define RX65N_USB_PHYSET_DIRPD                       (1U<<0)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:1352]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 37]
```diff
@@ -594,6 +594,855 @@
...
+#define RX65N_USB_PIPEBUF_SIZEMASK                   (0x1fU<<10)
+#define RX65N_USB_PIPEBUF_BUFNMBMASK                 (0xffU<<10)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:1354]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 38]
```diff
@@ -594,6 +594,855 @@
...
+#define RX65N_USB_PIPEBUF_SHIFT                      (10U)
+
+/* Possibly below are used for differentiating Control/ D0 or D1 pipe... */
+
+#define RX65N_USB_FIFO_D0                            (0UL)
+#define RX65N_USB_FIFO_D1                            (1UL)
+#define RX65N_USB_FIFO_C                             (2UL)
+#define RX65N_USB_DEVADD_UPPHUB_SHIFT                (11U)
+#define RX65N_USB_DEVADD_HUBPORT_SHIFT               (8U)
+#define RX65N_USB_USBMC_VDCEN                        (1U<<7)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:1364]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 39]
```diff
@@ -594,6 +594,855 @@
...
+
+/* Define Synergy HCOR command/status bitmaps.  */
+
+#define RX65N_USB_DCP                                (0)
+#define RX65N_USB_DCPCTR_DATA1                       (1U<<7)
+#define RX65N_USB_DCPCTR_DATA0                       (1U<<8)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:1370]: Ditto

File: arch/renesas/src/rx65n/rx65n_definitions.h [hunk 40]
```diff
@@ -594,6 +594,855 @@
...
+#define RX65N_USB_FIFO_READ_ERROR                    (6)
+#define RX65N_USB_FIFO_READ_OVER                     (7)
+#define RX65N_USB_ED_BRDY                            (0x00000001U)
+#define RX65N_USB_ED_NRDY                            (0x00000002U)
+#define RX65N_USB_ED_BEMP                            (0x00000004U)
+#define RX65N_USB_ED_EOFERR                          (0x00000010U)
+#define RX65N_USB_ED_SIGN                            (0x00000020U)
+#define RX65N_USB_ED_SACK                            (0x00000040U)
+#define RX65N_USB_ED_TIMEOUT                         (0x00000080U)
+#define RX65N_USB_LPSTS_SUSPENDM                     (1U<<14)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_definitions.h:1433]: Ditto

**Review by acassis — CHANGES_REQUESTED** [review]

File: arch/renesas/src/rx65n/rx65n_definitions.h (hunk 40 above)
Comment (anjana-tel) [reply arch/renesas/src/rx65n/rx65n_definitions.h]: Thank you for the comments.

Alignment errors are corrected in  b3857f3

**Pull request merged**